src/test/java/duke/tasks/TaskListTest.java

```diff
@@ -0,0 +1,219 @@
+package duke.tasks;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.Test;
+
+import duke.storage.Storage;
+
+public class TaskListTest {
+    private final String listHeader = "Here are the tasks in your list:\n";
+    private final String emptyList = "You have no tasks in your list.";
+
+    @Test
+    public void add_tasks_success() {
+        String fileName = "test_add.txt";
+        Storage storage = new Storage(fileName);
+        storage.save("");
+
+        StringBuilder expected = new StringBuilder(listHeader);
+        TaskList taskList = new TaskList(fileName);
+        taskList.add(TaskList.TaskType.TODO, "todo");
+        assertEquals(expected.append("1.[T][ ] todo").toString(), taskList.toString());
+
+        taskList.add(TaskList.TaskType.DEADLINE, "deadline", "2023-08-25 23:59");
+        expected.append("\n2.[D][ ] deadline (by: Fri 25 Aug 2023 23:59)");
+        assertEquals(expected.toString(), taskList.toString());
+
+        taskList.add(TaskList.TaskType.EVENT, "event", "2023-08-26 23:59", "2023-08-27 23:59");
+        expected.append("\n3.[E][ ] event (from: Sat 26 Aug 2023 23:59 to: Sun 27 Aug 2023 23:59)");
+        assertEquals(expected.toString(), taskList.toString());
+    }
+
+    @Test
+    public void add_noDescription_exceptionThrown() {
```

Comment:
Nit: possible to use less complex naming convention?

Reply:
I was following the format for naming of tests in the [Coding Standard](https://se-education.org/guides/conventions/java/intermediate.html):
>Underscores may be used in test method names using the following three part format `featureUnderTest_testScenario_expectedBehavior()`
>
>e.g. `sortList_emptyList_exceptionThrown()` `getMember_memberNotFount_nullReturned`